recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,567 @@
+        # self.requires("rssdk/x.x.x", transitive_headers=True)
+        # self.requires("metslib/x.x.x", transitive_headers=True)
+        # self.requires("openmp/system", transitive_headers=True)
+        # self.requires("opennurbs/x.x.x", transitive_headers=True)
+        # self.requires("poisson4/x.x.x", transitive_headers=True)
+
+    def package_id(self):
+        used_deps = self._all_ext_deps(self.info.options)
+        # Disable options that have no effect
+        all_opts = [opt for opt, value in self.info.options.items()]
+        for opt in all_opts:
+            if opt.startswith("with_") and opt.split("_", 1)[1] not in used_deps:
+                setattr(self.info.options, opt, False)
+
+    def validate(self):
+        enabled_components = self._enabled_components()
+        for component in sorted(enabled_components):
+            for dep in self._external_deps.get(component, []):
+                if not self._is_enabled(dep):
+                    raise ConanInvalidConfiguration(
+                        f"'with_{dep}=True' is required when '{component}' is enabled."
+                    )
+            for dep in self._internal_deps[component]:
+                if dep not in enabled_components:
+                    raise ConanInvalidConfiguration(
+                        f"'{dep}=True' is required when '{component}' is enabled."
+                    )
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["PCL_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["WITH_LIBUSB"] = self._is_enabled("libusb")
+        tc.cache_variables["WITH_OPENGL"] = self._is_enabled("opengl")
+        tc.cache_variables["WITH_OPENMP"] = self._is_enabled("openmp")
+        tc.cache_variables["WITH_PCAP"] = self._is_enabled("pcap")
+        tc.cache_variables["WITH_PNG"] = self._is_enabled("png")
+        tc.cache_variables["WITH_QHULL"] = self._is_enabled("qhull")
+        tc.cache_variables["WITH_QT"] = self._is_enabled("qt")
+        tc.cache_variables["WITH_VTK"] = self._is_enabled("vtk")
+        tc.cache_variables["WITH_CUDA"] = self._is_enabled("cuda")
+        tc.cache_variables["BUILD_CUDA"] = self._is_enabled("cuda")
+        tc.cache_variables["BUILD_GPU"] = self._is_enabled("cuda")
+        tc.cache_variables["WITH_SYSTEM_ZLIB"] = True
+        tc.cache_variables["PCL_ONLY_CORE_POINT_TYPES"] = True
```

Comment:
I think this should be an option. C3I can't build without this, yes, but that doesn't mean that others can't.